internal/spokes/spokes.go

```diff
@@ -141,8 +141,15 @@ func (r *spokesReceivePack) execute(ctx context.Context) error {
 	// We only need to perform the references discovery when we are not using the HTTP protocol or, if we are using it,
 	// we only run the discovery phase when the http-backend-info-refs/advertise-refs option has been set
 	if r.advertiseRefs || !r.statelessRPC {
-		if err := r.performReferenceDiscovery(ctx); err != nil {
-			return err
+		isolatedReferenceDiscovery := os.Getenv("GIT_SOCKSTAT_VAR_spokes_receive_pack_isolated_reference_discovery")
+		if isolatedReferenceDiscovery == "true" {
```

Comment:
Bools are encoded as `bool:<value>` in sockstat, so this comparison won't be quite right. https://github.com/github/spokes-receive-pack/pull/68 includes a fix (as well as a refactoring of sockstat code so that it has better abstractions).